python/cudf/cudf/core/column/string.py

```diff
@@ -652,26 +652,14 @@ def serialize(self):
         frames = []
         sub_headers = []
 
-        sbuf = rmm.device_array(self.nvstrings.byte_count(), dtype="int8")
-        obuf = rmm.device_array(len(self.nvstrings) + 1, dtype="int32")
-        mask_size = utils.calc_chunk_size(
-            len(self.nvstrings), utils.mask_bitsize
-        )
-        nbuf = rmm.device_array(mask_size, dtype="int8")
-        self.nvstrings.to_offsets(
-            libcudf.cudf.get_ctype_ptr(sbuf),
-            libcudf.cudf.get_ctype_ptr(obuf),
-            nbuf=libcudf.cudf.get_ctype_ptr(nbuf),
-            bdevmem=True,
-        )
-        for item in [sbuf, obuf]:
+        for item in reversed(self.children):
             sheader = item.__cuda_array_interface__.copy()
             sheader["dtype"] = item.dtype.str
             sub_headers.append(sheader)
             frames.append(item)
 
         if self.null_count > 0:
-            frames.append(nbuf)
+            frames.append(self.mask)
 
         header["nvstrings"] = len(self.nvstrings)
```

Comment:
Sounds like generating this could come with some overhead. Should we skip this in case it doesn't already exist?

Reply:
This will only have overhead if there's a non-zero offset. In that case we'd need to ship the entire mask, the entire data buffer, and the offset, which would likely be more expensive. I think this is what we want for now and we can adjust later if needed.

Reply:
Makes sense. Changed in the latest commit.

Reply:
Actually I may be missing something here. Couldn't we just do `len(self.children[0]) - 1`?

Reply:
Or do `len(self)` and update `__len__` to do what you mentioned :)

Reply:
Should add with the recent commit we actually don't need this any more. So will drop it.

Reply:
Sorry missed that comment. Yeah adding `len` is a good idea independently.

Reply:
Added PR ( https://github.com/rapidsai/cudf/pull/4113 ) for the `__len__` change 🙂